resolver_conn_wrapper_test.go

```diff
@@ -96,7 +98,7 @@ func TestDialParseTargetUnknownScheme(t *testing.T) {
 		// For known scheme.
 		{"passthrough://a.server.com/google.com", "google.com"},
 	} {
-		dialStrCh := make(chan string, 1)
+		dialStrCh := make(chan string)
```

Comment:
This should stay ", 1" and @menghanl's change to make this test more permissive still LGTM.  The new test you added will help us catch the race for regression purposes.

Reply:
Done